server/resource/caldav_service.go

```diff
@@ -0,0 +1,426 @@
+
+	actionRequestParameters := make(map[string]interface{})
+	actionRequestParameters["to"] = cs.Email
+	actionRequestParameters["subject"] = subject
+	actionRequestParameters["from"] = "daptin.no-reply"
+	actionRequestParameters["body"] = content
+
+
+	_, _, mailerError := cs.Mailer.DoAction(Outcome{}, actionRequestParameters)
+	if mailerError  != nil{
+		log.Error("Unable To Send mail", mailerError)
+	}
+
+	log.Info("resource updated ", rpath)
+	return &res, nil
+}
+
+func (cs *CalDavStorage) DeleteResource(rpath string) error {
+	a, err := cs.haveAccess(rpath, "admin")
+	if err != nil {
+		log.Error(err, "failed to get Access [" + rpath + "]")
+		return  err
+	}
+	if ! a {
+		log.Info("no access to collection [" + rpath + "]")
+		return nil
+	}
+
+	err = cs.cruds["calendar"].DeleteCalendarEvent(cs.UserID, rpath)
+	if err != nil {
+		log.Info("failed to delete resource ", rpath, " ", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (cs *CalDavStorage) isResourcePresent(rpath string) bool {
+	rrpath, err := cs.cruds["calendar"].GetRpath(cs.UserID)
+
+	if err != nil {
+		return false
+	}
+
+	if rrpath == rpath {
+		return true
+	}
+
+	return false
+}
+
+type PGResourceAdapter struct {
+	db map[string]*DbResource
+	resourcePath string
+	UserID       int64
+}
+
+func (pa *PGResourceAdapter) CalculateEtag() string {
+	if pa.IsCollection() {
+		return ""
+	}
+
+	return fmt.Sprintf(`"%x%x"`, pa.GetContentSize(), pa.GetModTime().UnixNano())
+}
+
+func (pa *PGResourceAdapter) haveAccess(perm string) (bool, error) {
+	permInst := pa.db[USER_ACCOUNT_TABLE_NAME].GetObjectPermissionById("calendar", pa.UserID)
+	uGroupId := permInst.UserGroupId
+
+	refId, err := pa.db[USER_ACCOUNT_TABLE_NAME].GetIdToReferenceId("calendar", pa.UserID)
+	if err != nil {
+		return false, err
+	}
+
+	if perm == "read"{
+		return permInst.CanRead(strconv.FormatInt(pa.UserID, 10), uGroupId), nil
+	}
+
+	if perm == "write"{
+		return permInst.CanCreate(strconv.FormatInt(pa.UserID, 10), uGroupId), nil
+	}
+
+	if perm == "admin"{
+		return pa.db[USER_ACCOUNT_TABLE_NAME].IsAdmin(refId), nil
+	}
+
+	return false, nil
+}
+
+func (pa *PGResourceAdapter) GetContent() string {
+	if pa.IsCollection() {
+		return ""
+	}
+	a, err := pa.haveAccess("read")
+	if err != nil {
+		log.Error(err, "failed to get Access ", pa.resourcePath)
+		return ""
+	}
+	if ! a {
+		return ""
+	}
+	content,err := pa.db["calendar"].GetContent(pa.resourcePath,pa.UserID)
+	if err != nil {
+		log.Error(err, "failed to fetch content ", pa.resourcePath)
+		return ""
+	}
+	ret, err := base64.StdEncoding.DecodeString(content)
+	if err != nil {
+		log.Error(err, "decode error ", pa.resourcePath)
+		return ""
+	}
+	return string(ret)
+}
+
+func (pa *PGResourceAdapter) GetContentSize() int64 {
+	return int64(len(pa.GetContent()))
```

Comment:
this would return size of the base64 encoded string and not the size of the actual content (i see you are decoding the value on line 391)